### DISCUSSION TITLE
Avalonia YoutubeDownload

### DISCUSSION THREAD

**Discussion by CarlosEX** [AvaloniaUI/Avalonia#5714] [General]
Hello Guys,

I'm a novice developer, but I'd like to share some of what I'm learning in this new world of programming with Avalonia.

This project consists of an application that will have 3 main modules.
1. Download (Youtube, Vímeo and Facebook)
2. Upload (Youtube...)
3. Artificial Intelligence and Extraction

This first step, I managed to implement the features to download video, audio, images and texts from youtube videos. And also the possibility to download me lots through the playlist.

https://user-images.githubusercontent.com/33513870/112534488-e3fd5b00-8d89-11eb-821a-fba340dd5b4b.mp4

Any suggestions will be welcome. I just need some help from you. I'll need to implement a video and audio feature to play the downloaded files. However, I don't know what would be more reliable, given that my environment is on Linux with Fedora, and my machine is a Dell with i3 and 4GB of RAM.

[Full video](https://youtu.be/-53uj1sHL9c)

**Comment by maxkatz6** [comment]
> I'll need to implement a video and audio feature to play the downloaded files.

There are two VLC wrappers for Avalonia:
https://github.com/videolan/libvlcsharp
https://github.com/donandren/vlcsharpavalonia

You should check them

**Reply by CarlosEX** [reply]
Thank you very much, I will implement them in my application. I'd just have another question. When I try to upload the image I downloaded from the video at runtime, but the image does not appear, only appears when I close and open the application again

**Reply by maxkatz6** [reply]
I think we need more details about image issue. Like code example.
You should be able to load images as a Bitmap source from the any stream, that can be read. So you can use .net FileInfo as well, or directly http stream/ memory stream.
If you try to use "avares" or "resx" path with downloaded images, it won't work, because it is part of compile time metadata.
